src/application/services/useEditor.ts

```diff
@@ -0,0 +1,173 @@
+import { onBeforeUnmount, onMounted, ref, watch } from 'vue';
+import Editor, { type OutputData, type API } from '@editorjs/editorjs';
+import type EditorTool from '@/domain/entities/EditorTool';
+// @ts-expect-error: we need to rewrite plugins to TS to get their types
+import Header from '@editorjs/header';
+import { useAppState } from './useAppState';
+
+/**
+ * Interface describing hook props
+ */
+interface EditorProps {
+    /**
+     * Content displayed in Editor
+     */
+    content?: OutputData;
+
+    /**
+     * Allow edit content in Editor
+     */
+    isReadOnly?: boolean;
+
+    /**
+     * Handler for every content change in Editor
+     */
+    onChange: (api: API) => void
+}
+
+/**
+ * Load one tool at a time
+ *
+ * @param src - source path to tool
+ */
+function loadScript(src: string): Promise<Event> {
+  return new Promise(function (resolve, reject) {
+    const script = document.createElement('script');
+
+    script.src = src;
+    script.onload = resolve;
+    script.onerror = reject;
+    document.head.appendChild(script);
+  });
+}
+
+/**
+ * Application service for working with Editor
+ */
+export function useEditor({ content, isReadOnly, onChange }: EditorProps): void {
+  /**
+   * Editor.js instance
+   */
+  const editor = ref<Editor | undefined>(undefined);
+
+  /**
+   * Editor content instance
+   */
+  const refContent = ref<OutputData | undefined>(content);
```

Comment:
same question

Reply:
```
watch(
    () => refContent,
    (watchedRef) => {
      const watchedContent = watchedRef.value;
    }
  );
  ```
  Isn't it? We update editor where content was updated